### ISSUE TITLE
Get meta data for output

### ISSUE THREAD

**Issue by NTShop** [savonet/liquidsoap#405]
I'm using webcast.js and webcaster to send a stream through Liquidsoap and on to Icecast. I want to save the stream to a file as it plays, and to do that I want to get the meta data sent by webcaster via webcast.js to use in formatting the out file name. I've tried to integrate on_metadata() and map_metadata() but so far haven't been able to figure out how to do it correctly. Here's my config .liq, can someone tell me how to intercept the meta data so I set the show_title string appropriately? 

	set( 'harbor.verbose', true )

	set( "log.file.path", "/tmp/liquidsoap")

	set( 'log.level', 5 )

	set( "harbor.bind_addr", "0.0.0.0" )

	live = input.harbor( 'live', port=9000, user="source", password="hackme")

	radio = fallback(track_sensitive=false, [live] )

	out = output.icecast( %mp3(bitrate=56, samplerate=48000, id3v2=true), radio, fallible=true, host="localhost", port=8000, user="source", password="hackme", mount="listen.mp3" )

	# timestamp = '%d-%m-%Y'

	# show_title = 'XXXXXXX'

	# output.file(%mp3(bitrate=32, samplerate=22050, id3v2=true), reopen_on_metadata=false, "/var/www/html/recorded_shows/#{show_title}-#{timestamp}.mp3", live, fallible=true)

**Comment by toots** [comment]
You have to use the interpolation API for string. It allows you to replace part of a string with the source's metadata. Here's the syntax:
```
filename = "/var/www/html/recorded_shows/$(show_title)-$(timestamp)-%m-%d-%Y-%H:%M:%S.mp3"
output.file(%mp3(bitrate=32, samplerate=22050, id3v2=true), 
                   reopen_on_metadata=true,live, fallible=true)
```

This will create a different file each time the metadata change. You will need to make sure that those metadata are present. You can always use an intermediate `map_metadata` operator to format them the way the output expects them.

Also, the filename should have date information added to it so you don't erase previous archives..

I'm keeping this open, online documentation should be updated to have more information about this.

**Comment by NTShop** [comment]
I saw in some example code how to do that. Maybe I didn't explain my issue clear enough, I'll try again: 

Using Webcaster/Webcast.js, I tweaked the code so that it automatically sends whatever meta data is in the settings area of Webcaster to liquidsoap as soon as a connection to liquidsoap is opened. 

In liquidsoap, somehow I want to intercept that metadata (sent from Webcaster/Webcast.js) and use it to assign a filename for output.file() to use in recording the stream. 

What I cannot figure out is how to intercept the metadata. Or maybe there's a better approach? Is it possible for me to send arbitrary data into liquidsoap over the input.harbor() websocket connection so that I can receive and parse via code in the start.liq script file that I use? If so how do I do that? Or is there some other way?  So far I'm at a loss to figure this out. I'm a long time developer, I understand PHP and JS coding very good, but I don't understand how to code for the liquidsoap startup script.

**Comment by NTShop** [comment]
I discovered that I can use server.register() to set up commands, but since this is on a remote server I cannot enable telnet since that's a risk, therefore I have to enable the Unix file socket and then write server side code to receive a command from the client and pass that command into the socket for processing. That works. (And this isn't exactly clear in the docs - the fact that it's a FILE based socket and therefore commands cannot be sent over the input.harbor socket connection (or at least when I tried it didn't work). 

Anyway, a suggestion:  It seems to me it'd be far better if we could enable some sort of HTTP or Websocket listener to handle sending commands / interacting and including some simple auth requirement on that listener.
